test/user.test.js

```diff
@@ -1267,6 +1267,109 @@ describe('User', function() {
     });
   });
 
+  describe('User.changePassword()', () => {
+    let userId, currentPassword;
+    beforeEach(givenUserIdAndPassword);
+
+    it('changes the password - callback-style', done => {
+      User.changePassword(userId, currentPassword, 'new password', (err) => {
+        if (err) return done(err);
+        expect(arguments.length, 'changePassword callback arguments length')
+          .to.be.at.most(1);
+
+        User.findById(userId, (err, user) => {
+          if (err) return done(err);
+          user.hasPassword('new password', (err, isMatch) => {
+            if (err) return done(err);
+            expect(isMatch, 'user has new password').to.be.true();
+            done();
+          });
+        });
+      });
+    });
+
+    it('changes the password - Promise-style', () => {
+      return User.changePassword(userId, currentPassword, 'new password')
+        .then(() => {
+          expect(arguments.length, 'changePassword promise resolution')
+            .to.equal(0);
+          return User.findById(userId);
+        })
+        .then(user => {
+          return user.hasPassword('new password');
+        })
+        .then(isMatch => {
+          expect(isMatch, 'user has new password').to.be.true();
+        });
+    });
+
+    it('fails when current password does not match', () => {
+      return User.changePassword(userId, 'bad password', 'new password').then(
+        success => { throw new Error('changePassword should have failed'); },
+        err => {
+          // workaround for chai problem
+          //   object tested must be an array, an object, or a string,
+          //    but error given
```

Comment:
FWIW, I have fixed the indentation in 4e60e62, I don't know why github did not recognize this comment as outdated.